brave-core/src/main/java/com/github/kristofa/brave/HttpSpanCollector.java

```diff
@@ -0,0 +1,198 @@
+   * @param metrics Gets notified when spans are accepted or dropped. If you are not interested in
+   * these events you can use {@linkplain EmptySpanCollectorMetricsHandler}
+   */
+  public static HttpSpanCollector create(String baseUrl, SpanCollectorMetricsHandler metrics) {
+    return new HttpSpanCollector(baseUrl, metrics, 1);
+  }
+
+  // Visible for testing. Ex when tests need to explicitly control flushing, set interval to 0.
+  HttpSpanCollector(String baseUrl, SpanCollectorMetricsHandler metrics, int flushInterval) {
+    this.url = baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "api/v1/spans";
+    this.metrics = metrics;
+    this.flusher = flushInterval > 0 ? new Flusher(this, flushInterval) : null;
+  }
+
+  /**
+   * Queues the span for collection, or drops it if the queue is full.
+   *
+   * @param span Span, should not be <code>null</code>.
+   */
+  @Override
+  public void collect(Span span) {
+    metrics.incrementAcceptedSpans(1);
+    if (!pending.offer(span)) {
+      metrics.incrementDroppedSpans(1);
+    }
+  }
+
+  /**
+   * Calling this will flush any pending spans to the http transport on the current thread.
+   */
+  @Override
+  public void flush() {
+    if (pending.isEmpty()) return;
+    Queue<Span> drained = new ArrayDeque<>(pending.size());
+    pending.drainTo(drained);
+    if (drained.isEmpty()) return;
+
+    // Thrift-encode the spans for transport
+    int spanCount = drained.size();
+    BufferTransport buffer = new BufferTransport();
+    try {
+      TBinaryProtocol oprot = new TBinaryProtocol(buffer);
+      oprot.writeListBegin(new TList(TType.STRUCT, drained.size()));
+      while (!drained.isEmpty()) {
+        drained.remove().write(oprot);
+      }
+      oprot.writeListEnd();
+    } catch (TException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+
+    // Compress the thrifts, as they are usually full of strings
+    Buffer gzippedThrifts = new Buffer();
+    try (GzipSink gzipSink = new GzipSink(gzippedThrifts)) {
+      gzipSink.write(buffer.buffer, buffer.buffer.size());
+    } catch (IOException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+
+    // Send the thrifts to the zipkin endpoint
+    try {
+      postSpans(gzippedThrifts);
+    } catch (IOException e) {
+      metrics.incrementDroppedSpans(spanCount);
+      return;
+    }
+  }
+
+  /** Calls flush on a fixed interval */
+  static final class Flusher implements Runnable {
+    final Flushable flushable;
+    final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+
+    Flusher(Flushable flushable, int flushInterval) {
+      this.flushable = flushable;
+      this.scheduler.scheduleWithFixedDelay(this, 0, flushInterval, SECONDS);
+    }
+
+    @Override
+    public void run() {
+      try {
+        flushable.flush();
+      } catch (IOException ignored) {
```

Comment:
ignored bc Flusher doesn't actually throw an IOE.. this accepts the implemented interface for testing purposes, and since that declares IOE, we have to catch it.